File: src/Atlasd/Battlenet/Protocols/Game/ChatCommands/ChatCommandContext.cs
using System.Collections.Generic;

namespace Atlasd.Battlenet.Protocols.Game
{
    class ChatCommandContext
    {
        public ChatCommand Command { get; protected set; }
        public Dictionary<string, string> Environment { get; protected set; }
        public GameState GameState { get; protected set; }

        public ChatCommandContext(ChatCommand command, Dictionary<string, string> environment, GameState gameState)
        {
            Command = command;
            Environment = environment;
            GameState = gameState;
        }
    }
}

File: src/AtlasV/Battlenet/Protocols/Game/ChatCommands/ChatCommandContext.vb
Imports System.Collections.Generic

Namespace AtlasV.Battlenet.Protocols.Game
    Class ChatCommandContext
        Public Property Command As ChatCommand
        Public Property Environment As Dictionary(Of String, String)
        Public Property GameState As GameState

        Public Sub New(ByVal varCommand As ChatCommand,
                       ByVal varEnvironment As Dictionary(Of String, String),
                       ByVal varGameState As GameState)
            Command = varCommand
            Environment = varEnvironment
            GameState = varGameState
        End Sub
    End Class
End Namespace